server/middleware/setUpAuthentication.ts

```diff
@@ -1,15 +1,43 @@
-import type { Router } from 'express'
-import express from 'express'
 import passport from 'passport'
 import flash from 'connect-flash'
+import { Router } from 'express'
+import { Strategy } from 'passport-oauth2'
 import config from '../config'
-import auth from '../authentication/auth'
+import tokenVerifier from '../data/tokenVerification'
 import { HmppsUser } from '../interfaces/hmppsUser'
+import generateOauthClientToken from '../authentication/clientCredentials'
 
-const router = express.Router()
+passport.serializeUser((user, done) => {
+  // Not used but required for Passport
+  done(null, user)
+})
 
-export default function setUpAuth(): Router {
-  auth.init()
+passport.deserializeUser((user, done) => {
+  // Not used but required for Passport
+  done(null, user as Express.User)
+})
+
+passport.use(
+  new Strategy(
+    {
+      authorizationURL: `${config.apis.hmppsAuth.externalUrl}/oauth/authorize`,
+      tokenURL: `${config.apis.hmppsAuth.url}/oauth/token`,
+      clientID: config.apis.hmppsAuth.authClientId,
+      clientSecret: config.apis.hmppsAuth.authClientSecret,
+      callbackURL: `${config.domain}/sign-in/callback`,
+      state: true,
+      customHeaders: { Authorization: generateOauthClientToken() },
+    },
+    (token, refreshToken, params, profile, done) => {
+      return done(null, { token, username: params.user_name, authSource: params.auth_source })
+    },
+  ),
+)
+
+export default function setupAuthentication() {
+  const router = Router()
+  const authUrl = config.apis.hmppsAuth.externalUrl
+  const authParameters = `client_id=${config.apis.hmppsAuth.authClientId}&redirect_uri=${config.domain}`
```

Comment:
did you move these for any reason? [it's generally good practice to declare variables close to where they're used](https://stackoverflow.com/a/33246061).

Reply:
I think I'm used to the second example given in there, that's just a personal preference of mine 😄 I'll shift these back

Reply:
Resolved in https://github.com/ministryofjustice/hmpps-template-typescript/pull/413/commits/0755dfd828172be677a0ac16272efca07c95ce71